src/elements/content-explorer/ShareDialog.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import Modal from 'react-modal';
+import noop from 'lodash/noop';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { Button, Text } from '@box/blueprint-web';
+
+import ShareAccessSelect from '../common/share-access-select';
+import { CLASS_MODAL_CONTENT, CLASS_MODAL_OVERLAY, CLASS_MODAL } from '../../constants';
+import type { BoxItem } from '../../common/types/core';
+
+import messages from '../common/messages';
+
+import './ShareDialog.scss';
+
+export interface ShareDialogProps {
+    appElement: HTMLElement;
+    canSetShareAccess: boolean;
+    isLoading: boolean;
+    isOpen: boolean;
+    item: BoxItem;
+    onCancel: () => {};
+    onShareAccessChange: () => {};
+    parentElement: HTMLElement;
+}
+
+const ShareDialog = ({
+    appElement,
+    canSetShareAccess,
+    isLoading,
+    isOpen,
+    item,
+    onCancel,
+    onShareAccessChange,
+    parentElement,
+}: ShareDialogProps) => {
+    const { formatMessage } = useIntl();
+    let textInput = null;
+
+    const copy = () => {
+        if (textInput instanceof HTMLInputElement) {
+            textInput.select();
+            document.execCommand('copy');
+        }
+    };
+
+    const { shared_link: sharedLink }: BoxItem = item;
+    const { url } = sharedLink || {
+        url: formatMessage(messages.shareDialogNone),
+    };
+
+    return (
+        <Modal
+            appElement={appElement}
+            className={CLASS_MODAL_CONTENT}
+            contentLabel={formatMessage(messages.shareDialogLabel)}
+            isOpen={isOpen}
+            onRequestClose={onCancel}
+            overlayClassName={CLASS_MODAL_OVERLAY}
+            parentSelector={() => parentElement}
+            portalClassName={`${CLASS_MODAL} be-modal-share`}
+        >
+            <div className="be-modal-content">
+                <Text as="label">
+                    <FormattedMessage tagName="div" {...messages.shareDialogText} />
```

Comment:
do we need `tagName` still? i assume it was originally added for some block level style but i'm not sure if it's necessary

Reply:
going to remove this that should be there